somatic/modules/scan.py

```diff
@@ -173,7 +173,12 @@ def run(self):
         # constants
         constants = []
         for constant_name in self.aqn.read('scan', 'constant names'):
-            units = 'wn'  # TODO: this is a hack, remove it
+            for hardware in all_hardwares:
+                if hardware.name == constant_name:
+                    units = hardware.units
+                    if wt.units.kind(units) == 'energy':
+                        units = 'wn'
+                    break
```

Comment:
Why is the `break` statement needed?

Reply:
strictly speaking, it's not _needed_. It is however, there because once it determines the units/finds the hardware, it is useless to continue searching. Everything else will fail the `if` condition.

It is an (admittedly minuscule) performance consideration.